plugins/fileservices/fileservices.cpp

```diff
@@ -2509,4 +2511,56 @@ FILESERVICES_API void FILESERVICES_CALL fsDfuPlusExec(ICodeContext * ctx,const c
     }
 }
 
+FILESERVICES_API const char * FILESERVICES_CALL fsGetEspURL()
+{
+    Owned<IConstEnvironment> daliEnv = openDaliEnvironment();
+    Owned<IPropertyTree> env = getEnvironmentTree(daliEnv);
+
+    if (env.get())
+    {
+        StringBuffer espURL;
+        StringBuffer espInstanceComputerName;
+        StringBuffer bindingProtocol;
+        StringBuffer xpath;
+        StringBuffer instanceAddress;
+        StringBuffer espServiceType;
+
+        Owned<IPropertyTreeIterator> espProcessIter = env->getElements("Software/EspProcess");
+        ForEach(*espProcessIter)
+        {
+            Owned<IPropertyTreeIterator> espBindingIter = espProcessIter->query().getElements("EspBinding");
+            ForEach(*espBindingIter)
+            {
+                espBindingIter->query().getProp("@service",espURL.clear());
+                xpath.setf("Software/EspService[@name=\"%s\"]/Properties/@type", espURL.str());
+
+                if(env->getProp(xpath.str(), espServiceType.clear()))
+                {
+                    if (!espServiceType.isEmpty() && (strieq(espServiceType.str(),"WsSMC")))
+                    {
+                        if (espBindingIter->query().getProp("@protocol",bindingProtocol.clear()))
+                        {
+                            Owned<IPropertyTreeIterator> espInstanceIter = espProcessIter->query().getElements("Instance");
+                            ForEach(*espInstanceIter)
+                            {
+                                if (espInstanceIter->query().getProp("@computer",espInstanceComputerName.clear()))
+                                {
+                                    xpath.setf("Hardware/Computer[@name=\"%s\"]/@netAddress",espInstanceComputerName.str());
+                                    if (env->getProp(xpath.str(),instanceAddress.clear()))
+                                    {
+                                        espURL.setf("%s://%s:%d", bindingProtocol.str(), fsfResolveHostName(instanceAddress.str()), espBindingIter->query().getPropInt("@port",8010)); // FileSpray seems to be fixed
```

Comment:
May be better not to resolve the hostname - since we're aiming to switch to allow unresolved hostnames throughout the system.

Reply:
The call to fsfResolveHostName() was done in order to correctly convert a "." instanceAddress value.  I will substitute explicit code to convert "." to 127.0.0.1.

Reply:
. is not the same as 127.0.0.1 - that would NOT be the correct fix.

Reply:
OK.  Can you point me to a more appropriate conversion routine?  Or should I call fsfResolveHostName() only if the value is "."?

Reply:
Is there a problem with leaving the . as . ? We should accept that anywhere that an ip is expected...

If you really wa'.'nt to resolve it, you can special case . but not to 127.0.0.1 - rather to the result of GetCachedHostName()

Reply:
Opting to call fsfResolveHostName() only when instanceAddress == "." as that seems to be the minimal correct fix.